### PR TITLE
[Test] Regression speed test for rgcn with neighbor sampling

### PR DESCRIPTION
**Pull request by classicsong**
## Checklist
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage
- [ ] Code is well-documented
- [x] To the my best knowledge, examples are either not affected by this change,
      or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR

### CODE REVIEW

File: benchmarks/benchmarks/model_speed/bench_rgcn_homogeneous_ns.py [hunk 1]
```diff
@@ -0,0 +1,375 @@
...
+        if blocks is None:
+            # full graph training
+            blocks = [self.g] * len(self.layers)
+        h = feats
+        for layer, block in zip(self.layers, blocks):
+            block = block.to(self.device)
+            h = layer(block, h, block.edata['etype'], block.edata['norm'])
+        return h
+
+class NeighborSampler:
```
Comment (jermainewang) [inline benchmarks/benchmarks/model_speed/bench_rgcn_homogeneous_ns.py:95]: Can we use DGL's NeighborSampler here?

File: benchmarks/benchmarks/model_speed/bench_rgcn_homogeneous_ns.py [hunk 2]
```diff
@@ -0,0 +1,375 @@
...
+    # Create csr/coo/csc formats before launching training processes with multi-gpu.
+    # This avoids creating certain formats in each sub-process, which saves momory and CPU.
+    g.create_formats_()
+
+    sampler = NeighborSampler(g, target_idx, fanouts)
+    loader = DataLoader(dataset=train_idx.numpy(),
+                        batch_size=batch_size,
+                        collate_fn=sampler.sample_blocks,
+                        shuffle=True,
+                        num_workers=num_workers)
```
Comment (jermainewang) [inline benchmarks/benchmarks/model_speed/bench_rgcn_homogeneous_ns.py:299]: Use DGL's NodeDataLoader?

File: benchmarks/benchmarks/model_speed/bench_rgcn_homogeneous_ns.py [hunk 3]
```diff
@@ -0,0 +1,375 @@
...
+                           layer_norm=False)
+
+    embed_layer = embed_layer.to(device)
+    model = model.to(device)
+
+    all_params = itertools.chain(model.parameters(), embed_layer.embeds.parameters())
+    optimizer = th.optim.Adam(all_params, lr=lr, weight_decay=l2norm)
+    emb_optimizer = th.optim.SparseAdam(list(embed_layer.node_embeds.parameters()), lr=lr)
+
+    for epoch in range(1):
```
Comment (jermainewang) [inline benchmarks/benchmarks/model_speed/bench_rgcn_homogeneous_ns.py:333]: How long does it take for one epoch?

**Review by jermainewang — APPROVED** [review]
Some minor comments. Otherwise LGTM.

File: benchmarks/benchmarks/model_speed/bench_rgcn_homogeneous_ns.py (hunk 3 above)
Comment (classicsong) [reply benchmarks/benchmarks/model_speed/bench_rgcn_homogeneous_ns.py:284]: 60 secs

**Comment by BarclayII** [comment]
Let me merge the two collators together.  I think it's possible to let `NodeDataLoader` and `EdgeDataLoader` to return the indices with an additional flag.

**Comment by BarclayII** [comment]
Done in #2495.

**Pull request merged**